image/helpers.go

```diff
@@ -53,11 +46,77 @@ type Store interface {
 	Assertion(assertType *asserts.AssertionType, primaryKey []string, user *auth.UserState) (asserts.Assertion, error)
 }
 
+// ToolingStore wraps access to the store for tools.
+type ToolingStore struct {
+	sto  Store
+	user *auth.UserState
+}
+
+func newToolingStore(arch, storeID string) (*ToolingStore, error) {
+	cfg := store.DefaultConfig()
+	cfg.Architecture = arch
+	cfg.StoreID = storeID
+	var user *auth.UserState
+	if authFn := os.Getenv("UBUNTU_STORE_AUTH"); authFn != "" {
+		var err error
+		user, err = readAuthFile(authFn)
+		if err != nil {
+			return nil, err
+		}
+	}
+	sto := store.New(cfg, nil)
+	return &ToolingStore{
+		sto:  sto,
+		user: user,
+	}, nil
+}
+
+type authData struct {
+	Macaroon   string   `json:"macaroon"`
+	Discharges []string `json:"discharges"`
+}
+
+func readAuthFile(authFn string) (*auth.UserState, error) {
+	f, err := os.Open(authFn)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open auth file %q: %v", authFn, err)
+	}
+	defer f.Close()
+
+	var user auth.UserState
```

Comment:
Tiny nit, but this is unused until the very end of the func, so maybe simplify a litte with `return &auth.UserState{...}, nil`?